shimmingtoolbox/shim/shim_utils.py

```diff
@@ -191,3 +191,114 @@ def shim_to_phys_cs(coefs, manufacturer):
+    """
+
+    scanner_shim = {
+        'f0': None,
+        'order1': None,
+        'order2': None,
+        'order3': None,
+        'has_valid_settings': False
+    }
+
+    # get_imaging_frequency
+    if bids_json_dict.get('ImagingFrequency'):
+        scanner_shim['f0'] = int(bids_json_dict.get('ImagingFrequency') * 1e6)
+
+    # get_shim_orders
+    if bids_json_dict.get('ShimSetting'):
+        scanner_shim['order1'] = bids_json_dict.get('ShimSetting')[:3]
+        scanner_shim['order2'] = bids_json_dict.get('ShimSetting')[3:]
+        scanner_shim['has_valid_settings'] = True
+    else:
+        logger.warning("ShimSetting tag is not available")
+
+    return scanner_shim
+
+
+def convert_to_mp(manufacturers_model_name, shim_settings):
+    """ Converts the ShimSettings tag from the json BIDS sidecar to the scanner units.
+        (i.e. For the Prisma fit DAC --> uT/m, uT/m^2 (1st order, 2nd order))
+
+    Args:
+        manufacturers_model_name (str): Name of the model of the scanner. Found in the json BIDS sidecar under
+                                        ManufacturersModelName'. Supported names: 'Prisma_fit'.
+        shim_settings (dict): Dictionnary with keys: 'order1', 'order2', 'has_valid_settings'. 'order1' is a list of 3
+                       coefficients for the first order. Found in the json BIDS sidecar under 'ShimSetting'. 'order2' is
+                       a list of 5 coefficients. 'has_valid_settings' is a boolean.
+
+    Returns:
+        dict: Same dictionnary as the shim_settings input with coefficients of the first, second and third order
+              converted according to the appropriate manufacturer model.
+    """
+    scanner_shim_mp = shim_settings
+
+    if manufacturers_model_name == "Prisma_fit":
+        if shim_settings.get('order1'):
+            # One can use the Siemens commandline AdjValidate tool to get all the values below:
+            max_current_mp_order1 = np.array([2300] * 3)
+            max_current_dcm_order1 = np.array([14436, 14265, 14045])
+            order1_mp = np.array(shim_settings['order1']) * max_current_mp_order1 / max_current_dcm_order1
+
+            if np.any(np.abs(order1_mp) > max_current_mp_order1):
+                scanner_shim_mp['has_valid_settings'] = False
+                raise ValueError("Multipole values exceed known system limits.")
+            else:
+                scanner_shim_mp['order1'] = order1_mp
+
+        if shim_settings.get('order2'):
+            max_current_mp_order2 = np.array([4959.01, 3551.29, 3503.299, 3551.29, 3487.302])
+            max_current_dcm_order2 = np.array([9998, 9998, 9998, 9998, 9998])
+            order2_mp = np.array(shim_settings['order2']) * max_current_mp_order2 / max_current_dcm_order2
+
+            if np.any(np.abs(order2_mp) > max_current_mp_order2):
+                scanner_shim_mp['has_valid_settings'] = False
+                raise ValueError("Multipole values exceed known system limits.")
```

Comment:
typo mistake + why are you saying multiples, in some case, there can be, just one value that is not respecting limits

Reply:
This is not a typo. Mulipole is the type of unit